lib/reek/report/report.rb

```diff
@@ -121,6 +123,41 @@ def total_smell_count_message
     end
 
     #
+    # Displays the status of each file as it is examined
+    # Displays the text report when complete
+    #
+    # @public
+    class ProgressReport < TextReport
```

Comment:
Also, please keep in mind that I've only spent about an hour looking at this code base, so if there is a better way to implement it I'm listening. 😄